Sources/TensorFlow/Operators/Math.swift

```diff
@@ -1706,35 +1708,73 @@ public extension Tensor where Scalar: Numeric & Comparable {
 }
 
 internal extension Tensor where Scalar: TensorFlowFloatingPoint {
-  @inlinable
-  func _vjpMinOrMax(squeezingAxes axes: Tensor<Int32>) -> (Tensor, (Tensor) -> Tensor) {
-    let result = max(squeezingAxes: axes)
-    return (result, { v in
-      let yUnsqueezed = result.expandingShape(at: axes.scalars.map { Int($0) })
-      let gradientUnsqueezed = v.expandingShape(at: axes.scalars.map { Int($0) })
+    // Note: adapted from `_MinOrMaxGrad`:
+    // https://github.com/tensorflow/tensorflow/blob/r2.1/tensorflow/python/ops/math_grad.py#L223.
+    @inlinable
+    func _vjpMinMaxHelper(
+        squeezingAxes axes: Tensor<Int32>,
+        originalValue: Tensor,
+        seed: Tensor
+    ) -> Tensor {
+        let yUnsqueezed = originalValue.expandingShape(at: axes.scalars.map { Int($0) })
+        let gradientUnsqueezed = seed.expandingShape(at: axes.scalars.map { Int($0) })
 
-      // Compute the number of selected (maximum or minimum) elements in each reduction dimension.
-      // If there are multiple minimum or maximum elements then the gradient will be divided between
-      // them.
-      let indicators = Tensor(yUnsqueezed .== self)
-      let selectedCount = indicators.sum(alongAxes: axes)
+        // Compute the number of selected (maximum or minimum) elements in each reduction dimension.
+        // If there are multiple minimum or maximum elements then the gradient will be divided
+        // between them.
+        let indicators = Tensor(yUnsqueezed .== self)
+        let selectedCount = indicators.sum(alongAxes: axes)
 
-      return gradientUnsqueezed.broadcasted(toShape: self.shapeTensor) * indicators / selectedCount
-    })
-  }
+        return gradientUnsqueezed.broadcasted(toShape: self.shapeTensor) * indicators / selectedCount
+    }
 
-  @inlinable
-  func _vjpMinOrMax(alongAxes axes: Tensor<Int32>) -> (Tensor, (Tensor) -> Tensor) {
-    let result = max(alongAxes: axes)
```

Comment:
Note: here's the issue in the old implementation.

`_vjpMinOrMax(alongAxes:)` is registered as the derivative of `min(alongAxes:)`, but it returns `let result = max(alongAxes: axes)` as the original result.